### PR TITLE
feat: Add next and previous hotkeys

### PR DESCRIPTION
**Pull request by bitchaser12**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
This provides the much needed missing support for going to previous and next tasks using keyboard hotkeys. The current process is impossible to use without a mouse when reviewing already annotated tasks and this pr will allow the re-reviewing process of tasks to be able to be an entire keyboard based operation which significantly speeds up the process
https://github.com/HumanSignal/label-studio/issues/2311#issuecomment-1900503110

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
Adds a hotkey for next and previous tasks

#### What is the current behavior?
No hotkey for next and previous existed

#### What libraries were added/updated?
N/A

#### Does this change affect performance?
No

#### Does this change affect security?
No

#### What alternative approaches were there?
N/A

#### What feature flags were used to cover this change?
feat

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
Frontend Annotation Process
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

**Comment by bitchaser12** [comment]
Built and tested it works as expected. Fulfills this request: https://github.com/HumanSignal/label-studio/issues/2311#issuecomment-1900503110

**Comment by bitchaser12** [comment]
Further testing revealed that the next and previous hotkeys when on the review page, aren't going to the expected next and previous images. I'm guessing the previous and next methods are not giving the correct result when in the review window but I don't know the code thoroughly enough to know exactly whats going on.

**Comment by bitchaser12** [comment]
I just included an alternate way to do this which is by modifying dm.focus-next and dm.focus-previous to always go to the next/previous item instead of just changing focus to the next/previous item (which seemingly doesn't do anything now). This could be an even simpler way to do this. One potential downside of it though is that approach wouldn't work for going previous and next on tasks when in the label multiple tasks labeling view once this is fixed: https://github.com/HumanSignal/label-studio/issues/5357 @jombooth 

Also that second approach as is (or what was there before it with the selecting) has a bug where it doesnt scroll the page down once it gets passed the choices on the page so it doesnt load new items and eventually can't continue going next because of it. Also if you manually scroll the page down it appears to go back to the first item on the page.

edit: I found the source of that second bug mentioned above. This hack works around it but this is where that bug occurs at:
https://github.com/HumanSignal/label-studio/pull/5350/files#diff-843ceacdc988004db8884789b11629a7f380644a2b587c921f950d15333a3df0

Regarding that first bug, its not too hard to workaround since if you use the previous button, itll go back to the previous one before it resets the page so its not a huge deal. (with the fix/workaround to the second bug)

**Comment by bitchaser12** [comment]
I've been reviewing existing annotations on my end using these changes and its been working well btw. (primarily using the focus-next and previous changes). The all-keyboard approach is over 2x as fast when reviewing data for me at least

**Review by Serhii-beep — APPROVED** [review]

**Comment by bitchaser12** [comment]
@Serhii-beep @jombooth just checking to see about getting this incorporated in some form? I'm still running on my local build of this branch and I can no longer upgrade label studio until this gets incorporated.

**Comment by amymrou** [comment]
/jira create
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9353316622)
> Jira issue [TRIAG-589](https://humansignal.atlassian.net/browse/TRIAG-589) is created

[TRIAG-589]: https://humansignal.atlassian.net/browse/TRIAG-589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: web/libs/datamanager/src/components/MainView/DataView/DataView.js
Comment (juliosgarbi) [inline web/libs/datamanager/src/components/MainView/DataView/DataView.js]: We shouldn't be removing these feature flags. This needs to be done with proper planning, and removing them here could break the whole application

File: web/libs/editor/src/core/settings/keymap.json
Comment (juliosgarbi) [inline web/libs/editor/src/core/settings/keymap.json]: I think we have a problem with the naming. We're not moving to the next and previous annotations; we're moving to the next and previous tasks. We need to fix the naming

File: web/libs/editor/src/stores/AppStore.js
Comment (juliosgarbi) [inline web/libs/editor/src/stores/AppStore.js]: We should fix the comments and the hotkeys.addNamed function, as it doesn't move through the annotation list but through the task list

**Review by juliosgarbi — CHANGES_REQUESTED** [review]
Thanks for the contribution. I have some comments before approval. Could you take a look at the comments and address or fix them if necessary?

**Comment by bitchaser12** [comment]
> Thanks for the contribution. I have some comments before approval. Could you take a look at the comments and address or fix them if necessary?

No problem. I've pushed changes to address the pr comments and merged and resolved conflicts.

**Comment by bitchaser12** [comment]
@juliosgarbi I've rebuilt and tested locally after the changes on my end and after merging with the latest from development and everything works correctly in testing.

**Comment by moritzwilksch** [comment]
amazing, looking forward to this landing! can we get someone to approve the CI run?

**Comment by robot-ci-heartex** [comment]
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

**Comment by EdjeElectronics** [comment]
Hey hey! Has this feature been merged and built yet? Next and previous hotkeys would be super handy!

**Comment by bitchaser12** [comment]
> Hey hey! Has this feature been merged and built yet? Next and previous hotkeys would be super handy!

The pr has been open for 10 months now but looks like they haven't gotten to merging it yet.

**Comment by robot-ci-heartex** [comment]
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

**Comment by robot-ci-heartex** [comment]
This PR was closed because it has been stalled for 10 days with no activity.

**Comment by dionzand** [comment]
What a shame this PR hasn't been merged..

**Pull request closed without merging**
